Rework docs on logging levels (#111143)
elastic/elasticsearch: docs/reference/setup/logging-config.asciidoc

[[configuring-logging-levels]]
=== Configuring logging levels

Log4J 2 log messages include a _level_ field. In order of increasing verbosity,

> the available levels are `FATAL`, `ERROR`, `WARN`, `INFO`, `DEBUG` or `TRACE`.

Author: :+1: makes sense

By default {es} includes all messages at levels `INFO`, `WARN`, `ERROR` and
`FATAL` in its logs, but filters out messages at levels `DEBUG` and `TRACE`.
This is the recommended configuration. Do not filter out messages at `INFO` or
higher log levels or else you may not be able to understand your cluster's
behaviour or troubleshoot common problems. Do not enable logging at levels
`DEBUG` or `TRACE` unless you are following instructions elsewhere in this
manual which call for more detailed logging, or you are an expert user who will

> be reading the {es} source code to determine the meaning of the logs.

Author: I know what you mean, but I'd rather keep this phrased more emphatically in terms of "do not do X" so I can copy/paste this into support cases on this topic.

Messages are logged by a hierarchy of loggers which matches the hierarchy of
Java packages and classes in the {es-repo}[{es} source code]. Every logger has
a corresponding <<cluster-update-settings,dynamic setting>> which can be used
to control the verbosity of its logs. The setting's name is the fully-qualified

> name of the package or class, prefixed with `logger.`. For example, the
> `org.elasticsearch.discovery` package contains functionality related to the
> <<discovery-hosts-providers,discovery>> process, and you can control the
> verbosity of its logs with the `logger.org.elasticsearch.discovery` dynamic
> setting. You may set each logger's setting to the name of a log level, for
> instance `DEBUG`, which means that messages from this logger at levels up to
> the specified one will be included in the logs. Each logger can also be set to
> `OFF` to suppress its messages entirely.

Reviewer: ```suggestion
name of the package or class, prefixed with `logger.`. 

For example, the `org.elasticsearch.discovery` package contains functionality
related to the <<discovery-hosts-providers,discovery>> process. Adjust log
verbosity using `logger.org.elasticsearch.discovery` dynamic setting. Set the 
log level, for example`DEBUG`, to include all messages up to that level. 
To suppress all messages, set the logger to `OFF`.
```

Reviewer: suggestion to separate block of text and some minor edits for concision

Author: Split it up and reordered a bit but largely kept the original text as it's a little more precise.
